bundles/core/org.eclipse.smarthome.core.audio/src/main/java/org/eclipse/smarthome/core/audio/AudioFormat.java

```diff
@@ -238,6 +240,132 @@ public boolean isCompatible(AudioFormat audioFormat) {
         return true;
     }
 
+    /**
+     * Determines the best match between a list of audio formats supported by a source and a sink.
+     *
+     * @param inputs the supported audio formats of an audio source
+     * @param outputs the supported audio formats of an audio sink
+     * @return the best matching format or null, if source and sink are incompatible
+     */
+    public static AudioFormat getBestMatch(Set<AudioFormat> inputs, Set<AudioFormat> outputs) {
+        AudioFormat preferredFormat = getPreferredFormat(inputs);
```

Comment:
getPrefferedFormat can return null. Make it sense to look all the outputs (what is the intention of `output.isCompatible(null)`)?